code/datums/datumvars.dm

```diff
@@ -32,21 +32,9 @@
 	switch(var_name)
 		if("vars")
 			return FALSE
-		if("var_edited")
+		if(TRAIT_VAR_EDITED)
 			return FALSE
-	var_edited = TRUE
-	vars[var_name] = var_value
-
-	. = TRUE
-
-
-/client/vv_edit_var(var_name, var_value) //called whenever a var is edited
-	switch(var_name)
-		if("vars")
-			return FALSE
-		if("var_edited")
-			return FALSE
-	var_edited = TRUE
+	ADD_TRAIT(src, TRAIT_VAR_EDITED, TRAIT_SOURCE_ADMIN)
```

Comment:
I think for these low-level things for each datum, we should do what TG did and apply them to a datum_flags var. That is the best way to protect them from stuff like this, but also allows for cutting down on the amount of datum vars

Reply:
potentially? But it runs into the weird situation of "What if we want some stuff edited and we don't want others edited".

Reply:
I think all cases of stuff that TG's added to their datum vars would be safe to add to ours, i.e. var edited, is processing, etc. I dont think we ever want people messing around with datum vars really anyways